TestingLowerBounds/FDiv/CondFDiv.lean

```diff
@@ -685,6 +685,220 @@ lemma fDiv_compProd_right [CountablyGenerated β]
     · rw [← ne_eq, fDiv_ne_top_iff] at h_top
       exact h_top.1
 
+variable {γ : Type*} [MeasurableSpace γ]
+
+lemma condFDiv_kernel_snd'_integrable_iff [CountablyGenerated γ] [IsFiniteMeasure μ]
+    {ξ : kernel α β}  [IsFiniteKernel ξ] {κ η : kernel (α × β) γ} [IsMarkovKernel κ]
+    [IsMarkovKernel η] (h_ac : derivAtTop f = ⊤ → ∀ᵐ a ∂μ, ∀ᵐ b ∂ξ a, κ (a, b) ≪ η (a, b))
+    (h_int : ∀ᵐ a ∂μ, ∀ᵐ b ∂ξ a, Integrable (fun x ↦ f ((∂κ (a, b)/∂η (a, b)) x).toReal) (η (a, b)))
+    (h_int2 : ∀ᵐ a ∂μ, Integrable (fun b ↦ ∫ x, f ((∂κ (a, b)/∂η (a, b)) x).toReal ∂η (a, b)) (ξ a))
+    (hf_meas : StronglyMeasurable f) (hf_cvx : ConvexOn ℝ (Set.Ici 0) f)
+    (hf_cont : ContinuousOn f (Set.Ici 0)) (hf_one : f 1 = 0) :
+    Integrable (fun a ↦ (condFDiv f (kernel.snd' κ a) (kernel.snd' η a) (ξ a)).toReal) μ ↔
+      Integrable (fun a ↦ ∫ b, |∫ x, f ((∂κ (a, b)/∂η (a, b)) x).toReal ∂η (a, b)| ∂ξ a) μ := by
+  have h_ae_eq :
+      (fun x ↦ EReal.toReal (condFDiv f (kernel.snd' κ x) (kernel.snd' η x) (ξ x)))
+        =ᵐ[μ] fun a ↦ ∫ b, (fDiv f (κ (a, b)) (η (a, b))).toReal ∂ξ a := by
+    simp_rw [← kernel.snd'_apply] at h_int2 h_int h_ac ⊢
+    rw [← eventually_all] at h_ac
+    filter_upwards [h_ac, h_int, h_int2] with a ha_ac ha_int ha_int2
+    rw [condFDiv_eq ha_int ha_int2 ha_ac, EReal.toReal_coe]
+  have h_ereal_add : ∀ᵐ a ∂μ, ∀ᵐ b ∂ξ a, ∫ x, f ((∂κ (a, b)/∂η (a, b)) x).toReal ∂η (a, b)
+      + (derivAtTop f * ((κ (a, b)).singularPart (η (a, b)) Set.univ)).toReal
+      = (fDiv f (κ (a, b)) (η (a, b))).toReal := by
```

Comment:
Please make a new lemma for this as well (and reverse the equality). Also `(derivAtTop f).toReal * ((κ (a, b)).singularPart (η (a, b)) Set.univ).toReal` could be better than `(derivAtTop f * ((κ (a, b)).singularPart (η (a, b)) Set.univ)).toReal`? It looks like when you apply the lemma the first thing you do is apply `EReal.toReal_mul`.